collect_app/src/main/java/org/odk/collect/android/formmanagement/FormMediaDownloader.kt

```diff
@@ -0,0 +1,93 @@
+package org.odk.collect.android.formmanagement
+
+import org.odk.collect.android.utilities.FileUtils.constructMediaPath
+import org.odk.collect.android.utilities.FileUtils.copyFile
+import org.odk.collect.android.utilities.FileUtils.interuptablyWriteFile
+import org.odk.collect.async.OngoingWorkListener
+import org.odk.collect.forms.Form
+import org.odk.collect.forms.FormSource
+import org.odk.collect.forms.FormSourceException
+import org.odk.collect.forms.FormsRepository
+import org.odk.collect.forms.MediaFile
+import org.odk.collect.shared.strings.Md5.getMd5Hash
+import timber.log.Timber
+import java.io.File
+import java.io.IOException
+
+class FormMediaDownloader constructor(
```

Comment:
We're getting to a place where it feels like the best strategy for `ServerFormDownloader` is going to be breaking it into several components that we can just chain together in a "pipeline". Looking at this file in isolation, it's much easier to understand media file downloading that when it's surrounded by the whole form download process's logic.